### ISSUE TITLE
Cask request: SafeSign Identity Client

### ISSUE THREAD

**Issue by rogeriopradoj** [caskroom/homebrew-cask#24722]
### Cask details

**Name** - SafeSign Identity Client - *tokenadmin.app* and *SafeSignIC Tokenlounge.app*

**Homepage** - https://www.aeteurope.com/our-solutions/safesign-identity-client/

**License** - :gratis

**Download URL v3.0.112** - https://safeweb.com.br/portals/10/downloads/Safesign_3.0.112_MAC.zip

**Download URL v3.0.93** - http://site.solutinet.com.br/2014/images/files/instaladores-e-manuais/Safesign-3093-MacOS-driver-manual.zip

**Description** - middleware for strong authentication with smart cards and USB tokens.

**Comment by berniechiu** [comment]
I would like to add this one for hacktoberfest

**Comment by reitermarkus** [comment]
@berniechiu, go right ahead, here's some documentation to get you started: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md

**Comment by rogeriopradoj** [comment]
Hi, @berniechiu . I already have a skeleton here of this cask (I started the work on that when I opened the issue), I will share it on https://github.com/rogeriopradoj/homebrew-cask/tree/new-cask_safesign.

It cask is important for me because I'm using Smart Card Certificate  from [ICP-Brasil](http://www.iti.gov.br/icp-brasil), the PKI - Public Key Infrastructure of Brazilian Government.

On your questions:

> Q: (1) Should I specify two casks for this? Because I see two different urls being provided here. (make client cask depend on the driver cask? Not sure if this is provided in the DSL)

A: The two urls provided are for different versions (v3.0.112 and v3.0.93). Anyway I think that it should have 2 casks:
- one cask for **SafeSign Identity Client Standard - safesign-ic-standard** (that installs in a row 2 things: **Token admin utility** and PKCS#11 Lib (libaetpkss.dylib) plus Java Lib (libaetjcss.dylib)
- *if possible*, another cask for **SafeSign Identity Client TokenLounge - safesign-ic-tokenlounge** (provides all functionality necessary to use hardware tokens with applications that make use of the Apple Keychain)

**Note:** I say *if possible* because TokenLounge wasn't released for a long time because of incompatibilities with some versions of Mac OS X.
> Due to changes in the Mac OS X architecture AET was forced to discontinue its TokenLounge product for native applications supporting Keychain (such as Safari), as the Mac OS X architecture no longer offers support for products such as TokenLounge.
[Source: SafeSign v3.0.77 manual](http://www.downloadcertisign.com.br/site/midias/gerenciadores/safesign/Macintosh/SafeSign_3.0.77_Product_Description.pdf)

According to [AET](https://www.aeteurope.com/support/faq/faq-tokenlounge/) TokenLounge is back in Yosemite and El Capitan. But, according to my tests, with my personal macOS Sierra it's not working anymore*.

The products version v3.0.112 is inconsistent in macOS Sierra:
- [X] safesign-ic-standard: Token admin utility works fine
- [] safesign-ic-standard: PKCS#11 Lib cannot be added on Firefox or Thunderbird
- [] safesign-ic-tokenlounge: TokenLounge.app crashes when opened

The products ver v3.0.93 are OK in macOS Sierra:
- [x] safesign-ic-standard: Token admin utility works fine
- [x] safesign-ic-standard: PKCS#11 Lib can be added on Firefox or Thunderbird
* TokenLounge wasn't available in v3.0.93.

So, my suggestion for now is:
- new cask: safesign-ic-standard - v3.0.93

In near future, when AET (or its partners, issuers of ICP-Brasil digital certificates, as [Safeweb](https://safeweb.com.br), [Soluti](https://site.solutinet.com.br/2015/), [Certisign](https://www.certisign.com.br), the Federal Bank [Caixa Econômica Federal](http://www.caixa.gov.br/empresa/identidade-digital/Paginas/default.aspx) or any of this list by [Receita Federal do Brasil](http://idg.receita.fazenda.gov.br/orientacao/tributaria/senhas-e-procuracoes/senhas/certificados-digitais/orientacoes-sobre-emissao-renovacao-e-revogacao-de-certificados-digitais-e-cpf-ou-e-cnpj)) provides an updated version of SafeSign supporting macOS Sierra:
- update cask: safesign-ic-standard - vX.Y.Z
- new cask: safesign-ic-tokenlounge - vX.Y.Z


> (2) The first download url provided with zip and that seems alright, but while I unzipped to check it. I see two .dmg files. The documentation only show that the .app should present when unzip.

A: As only the second url, v3.0.93, is OK with macOS Sierra, I suggest we work only on that, ok? Anyway, when there's a zip and a dmg, you could use `container nested: ` in Formula.rb

In short: https://github.com/rogeriopradoj/homebrew-cask/tree/new-cask_safesign will have a safesign-ic-standard.rb

See ya!

**Comment by vitorgalvao** [comment]
I’m not sure this is a good fit for HBC. It seems to be relatively niche with an interested audience (for the cask) of one. It also seems to be fairly security minded, and it’s not like we’re really going to stay on top of it, or anything (most cask update contributions are from users).

This seems like the perfect candidate for a user [tap](https://github.com/Homebrew/brew/blob/master/docs/brew-tap.md). Alternatively, this might be a fit for [caskroom/eid](https://github.com/caskroom/homebrew-eid) (not sure, is it?).

Either way, closing the issue. If you still want to submit to the main project, we’ll surely review and discuss its inclusion at the time, but since you’re already working on the cask and seem to know a lot both about the software itself and how to contribute to HBC, I don’t see a point to have this issue open.

Cask requests are for when you need someone else to make the cask. If you’re going to work on it yourself, please don’t open an issue, just submit the PR.

**Comment by berniechiu** [comment]
> A: As only the second url, v3.0.93, is OK with macOS Sierra, I suggest we work only on that, ok? 

As @victorpopkov suggested, you think you can submit your PR. I'll work on the second one 💯

**Comment by rogeriopradoj** [comment]
Hi, @vitorgalvao :

> Alternatively, this might be a fit for caskroom/eid (not sure, is it?).

Brazil government is working for a Electronic ID but [it's not available yet](https://www.epochtimes.com.br/camara-e-senado-devem-votar-identificacao-civil-nacional-logo-apos-impeachment) (RCI, RCU and RCN).

Anyway, as SafeSign apps are used by many users of smart card readers and tokens for digital certificates (including Brazilian ICP-Brazil but not limited to them), I still think that the cask is fine here in HBC. So, let's discuss in PR.

Thanks again!

**Comment by andersonfreitas** [comment]
Hi @rogeriopradoj 
Is there any update on this? I'm installing the Safesign IC on my macOS Sierra and would love to use this Cask.
Thanks.
